src/NorthwindWebsite/Configuration/ServicesConfiguration.cs

```diff
@@ -101,11 +103,19 @@ public static class ServicesConfiguration
             options.DefaultSignInScheme = IdentityConstants.ExternalScheme;
             options.DefaultSignOutScheme = IdentityConstants.TwoFactorUserIdScheme;
         })
-            .AddCookie(CookieAuthenticationDefaults.AuthenticationScheme)
             .AddCookie(IdentityConstants.ApplicationScheme)
             .AddCookie(IdentityConstants.TwoFactorUserIdScheme)
             .AddExternalCookie();
 
+        services.AddAuthentication().AddMicrosoftAccount(microsoftOptions =>
+        {
+            microsoftOptions.ClientId = configuration["Authentication:Microsoft:ClientId"];
```

Comment:
Why do you extract it from `IConfiguration` if you pass here `AppSettings` that should be already initialised?